bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -139,6 +160,39 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
         }
     }
 
+    private Collection<Resource> getAssetFolders(Page page, ResourceResolver resolver) {
+        Map<String, Resource> allAssetFolders = new TreeMap<String, Resource>();
+        ValueMap properties = page.getProperties();
+        String[] configuredAssetFolderPaths = properties.get(damAssetProperty, String[].class);
+        if (configuredAssetFolderPaths != null) {
+            for (String configuredAssetFolderPath : configuredAssetFolderPaths) {
+                if (StringUtils.isNotBlank(configuredAssetFolderPath)) {
+                    Resource configuredAssetFolder = resolver.getResource(configuredAssetFolderPath);
+                    if (configuredAssetFolder != null && configuredAssetFolder.isResourceType("sling:OrderedFolder")) {
+                        for (Resource assetFolder : getAssetFolders(configuredAssetFolder)) {
+                            if (!allAssetFolders.containsKey(assetFolder.getPath())) {
+                                allAssetFolders.put(assetFolder.getPath(), assetFolder);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return allAssetFolders.values();
+    }
+
+    private List<Resource> getAssetFolders(Resource assetFolder) {
+        List<Resource> damFolders = new ArrayList<Resource>();
+        damFolders.add(assetFolder);
+        for (Iterator<Resource> children = assetFolder.listChildren(); children.hasNext();) {
+            Resource assetFolderChild = children.next();
+            if (assetFolderChild.isResourceType("sling:OrderedFolder")) {
```

Comment:
Duplicates? is this if there are multiple Dam Paths configures with one being under another .. ex.

For instance, do you mean if the sitemap was configured with 2 DAM Paths:

* /content/dam/demo/documents/brochures
* /content/dam/demo/documents

I expect it would be quicker to do the check on the list of paths to traverse first. In this case, you wouldnt traverse `/content/dam/demo/documents/brochures` since you have a ancestor rule to traverse `/content/dam/demo/documents`

I just worry that these traversals could get expensive on large DAMs, so would be nice to minimize as much as possible.